descartes_trajectory/include/descartes_trajectory/cartesian_interpolator.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2014, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTESIAN_INTERPOLATOR_H_
+#define CARTESIAN_INTERPOLATOR_H_
+
+#include <descartes_trajectory/cart_trajectory_pt.h>
+#include <tf/transform_datatypes.h>
+
+namespace descartes_trajectory
+{
+
+class TrajectorySegment
+{
+public:
+
+  virtual ~TrajectorySegment()
+  {
+
+  }
+
+  virtual bool interpolate(std::vector<descartes_core::TrajectoryPtPtr>& segment_points) = 0;
+
+protected:
+
+  TrajectorySegment(){}
+
+};
+
+class BlendSegment: public TrajectorySegment
+{
+public:
+
+  BlendSegment(const descartes_trajectory::CartTrajectoryPt& start,
+               const descartes_trajectory::CartTrajectoryPt& end,
+               double duration,
```

Comment:
The TrajectoryPt has a transition pointer which only imposes motion constraints but no time value as far as I know.  Plus judging from previous discussions (see #10) these where going to be replaced eventually.